lib/parser/calculate_size.js

```diff
@@ -94,7 +94,7 @@ function calculateElement(name, value, serializeFunctions, isArray, ignoreUndefi
         value['_bsontype'] === 'MaxKey'
       ) {
         return (name != null ? Buffer.byteLength(name, 'utf8') + 1 : 0) + 1;
-      } else if (value instanceof ObjectId || value['_bsontype'] === 'ObjectId') {
+      } else if (value instanceof ObjectId || value['_bsontype'] === 'ObjectId' || value['_bsontype'] === 'ObjectID') {
```

Comment:
I think we should remove all `instanceof` checks for BSON types, and depend strictly on the `_bsontype` field.

Reply:
Done.